jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriter.java

```diff
@@ -0,0 +1,307 @@
+	}
+
+	@Override
+	public void validateSetup(Server server, Query query) throws ValidationException {
+		// Not implemented
+	}
+
+	/**
+	 * <p>
+	 * Each {@link Result} is written as a {@link Point} to InfluxDB
+	 * </p>
+	 * 
+	 * <p>
+	 * The measurement for the {@link Point} is to {@link Result#getKeyAlias()}
+	 * <p>
+	 * <a href=
+	 * "https://influxdb.com/docs/v0.9/concepts/key_concepts.html#retention-policy">
+	 * The retention policy</a> for the measurement is set to "default" unless
+	 * overridden in settings:
+	 * </p>
+	 * 
+	 * <p>
+	 * The write consistency level defaults to "ALL" unless overridden in
+	 * settings:
+	 * 
+	 * <ul>
+	 * <li>ALL = Write succeeds only if write reached all cluster members.</li>
+	 * <li>ANY = Write succeeds if write reached any cluster members.</li>
+	 * <li>ONE = Write succeeds if write reached at least one cluster members.
+	 * </li>
+	 * <li>QUORUM = Write succeeds only if write reached a quorum of cluster
+	 * members.</li>
+	 * </ul>
+	 * 
+	 * <p>
+	 * The time key for the {@link Point} is set to {@link Result#getEpoch()}
+	 * </p>
+	 * 
+	 * <p>
+	 * All {@link Result#getValues()} are written as fields to the {@link Point}
+	 * </p>
+	 * 
+	 * <p>
+	 * The following properties from {@link Result} are written as tags to the
+	 * {@link Point} unless overriden in settings:
+	 * 
+	 * <ul>
+	 * <li>{@link Result#getAttributeName()}</li>
+	 * <li>{@link Result#getClassName()}</li>
+	 * <li>{@link Result#getObjDomain()}</li>
+	 * <li>{@link Result#getTypeName()}</li>
+	 * </ul>
+	 * <p>
+	 * {@link Server#getHost()} is set as a tag on every {@link Point}
+	 * </p>
+	 * 
+	 */
+	@Override
+	protected void internalWrite(Server server, Query query, ImmutableList<Result> results) throws Exception {
+		// Creates only if it doesn't already exist
+		influxDB.createDatabase(database);
+
+		BatchPoints batchPoints = BatchPoints.database(database).retentionPolicy(retentionPolicy)
+				.tag(TAG_HOSTNAME, server.getHost()).consistency(writeConsistency).build();
+		for (Result result : results) {
+			Map<String, String> resultTagsToApply = buildResultTagMap(result);
+			Point point = Point.measurement(result.getKeyAlias()).time(result.getEpoch(), TimeUnit.MILLISECONDS)
+					.tag(resultTagsToApply).fields(result.getValues()).build();
+			batchPoints.point(point);
+		}
+		influxDB.write(batchPoints);
+	}
+	
+	/**
+	 * Adds data from {@link Result} to a map based on the attributes configured in <code>resultAttributesToWriteAsTags</code>
+	 * @param result The {@link Result} to get the data from 
+	 * @return A map based on the attributes configured in <code>resultAttributesToWriteAsTags</code>
+	 * @throws Exception If refection cannot be performed on the {@link Result}
+	 */
+	private Map<String, String> buildResultTagMap(Result result) throws Exception {
+		Map<String, String> resultTagsToApply = new TreeMap<String, String>();
+		for (ResultAttribute resultAttribute : resultAttributesToWriteAsTags) {
```

Comment:
Nice usage of reflection! I'm wondering if this isn't too smart... It will at least break automatic refactoring of `Result` class. So we should at least externalize the `ResultAttribute` class and provide a unit test that validates that all possible properties can be extracted. Also, I'd like to have the extraction logic moved to the `ResultAttribute` class. This would give it an actual responsibility and it would act as more than a dumb data object.